flaml/autogen/agentchat/responsive_agent.py

```diff
@@ -891,6 +907,17 @@ def execute_code_blocks(self, code_blocks):
                 )
                 # raise NotImplementedError
             self._code_execution_config["use_docker"] = image
+
+            if (
+                self._code_execution_config.get("output_limit", -1) > 0
+                and count_token(logs_all + "\n" + logs) > self._code_execution_config["output_limit"]
+            ):
+                if self._code_execution_config.get("on_output_limit", None) is not None:
+                    logs_all = self._code_execution_config["on_output_limit"](logs_all, logs)  # replace the whole log
+                else:
+                    logs_all += "\n" + "Error: The output exceeds the length limit and is truncated."
```

Comment:
after this the logs could still be longer than the length limit.

Reply:
I added doc for this, explaining that the default handling could exceed token_limit at 13 more tokens. I think this works because this feature is to prevent unexpected long outputs, instead of picking on a few tokens.